### PR TITLE
build!: update to latest version of gts and typescript

### PR DESCRIPTION
**Pull request by sofisl**

### CODE REVIEW

File: .prettierignore [hunk 1]
```diff
@@ -1,3 +1,5 @@
-node_modules/*
-samples/node_modules/*
-src/**/doc/*
+**/node_modules
+**/.coverage
+build/
+docs/
+protos/
```
Comment (JustinBeckwith) [inline .prettierignore:5]: Please add a newline at the end to get rid of the squiggle

File: package.json [hunk 2]
```diff
@@ -54,14 +54,17 @@
     "eslint-config-prettier": "^6.0.0",
     "eslint-plugin-node": "^11.0.0",
     "eslint-plugin-prettier": "^3.0.0",
+    "gts": "^2.0.0-alpha.5",
```
Comment (JustinBeckwith) [inline package.json:57]: We can drop the `-alpha.5` part here since `2.0` is out

File: package.json [hunk 3]
```diff
@@ -54,14 +54,17 @@
     "eslint-config-prettier": "^6.0.0",
```
Comment (JustinBeckwith) [inline package.json:54]: Please delete all the `eslint-` packages, and the `prettier` package

File: package.json [hunk 4]
```diff
@@ -54,14 +54,17 @@
...
     "eslint-plugin-prettier": "^3.0.0",
+    "gts": "^2.0.0-alpha.5",
     "jsdoc": "^3.6.2",
     "jsdoc-fresh": "^1.0.1",
     "jsdoc-region-tag": "^1.0.2",
     "linkinator": "^2.0.0",
-    "mocha": "^7.0.0",
+    "mocha": "^7.1.1",
     "c8": "^7.0.0",
+    "chai": "^4.2.0",
```
Comment (JustinBeckwith) [inline package.json:64]: Do we need chai here?

File: package.json (hunk 3 above)
Comment (sofisl) [reply package.json:54]: Hey @JustinBeckwith! Just want to make sure I'm understanding correctly. Did you want me to delete .eslintignore, .eslintrc.json, .prettierignore and .prettierrc.js? I was using [this PR](googleapis/google-cloud-node/issues/2972
) as reference, I know Ben left those there.

Comment (JustinBeckwith) [reply package.json:54]: Oh noes!  Leave those files.  I'm saying the package references in the `devDependencies` section of `package.json`.  Those are transitive dependencies of `gts`, so we can delete em from here. 

Keep all the files!

**Pull request merged**